c10/util/intrusive_ptr.h

```diff
@@ -686,23 +687,14 @@ using weak_intrusive_ptr_target = intrusive_ptr_target;
 namespace raw {
 
 namespace intrusive_ptr {
-
-  // WARNING: Unlike the reclaim() API, it is NOT valid to pass
-  // NullType::singleton to this function
   inline void incref(intrusive_ptr_target* self) {
-    auto ptr = c10::intrusive_ptr<intrusive_ptr_target>::reclaim(self);
-    auto ptr_copy = ptr;
-    ptr_copy.release();
-    ptr.release();
+    if (self) {
+      ++self->refcount_;
+    }
   }
 
-  // WARNING: Unlike the reclaim() API, it is NOT valid to pass
```

Comment:
Is it now valid to pass in `NullType::singleton`? Otherwise maybe we should keep this comment